src/Components/Web/src/WebRenderer.cs

```diff
@@ -40,9 +41,15 @@ public abstract class WebRenderer : Renderer
 
         // Supply a DotNetObjectReference to JS that it can use to call us back for events etc.
         jsComponentInterop.AttachToRenderer(this);
+
         var jsRuntime = serviceProvider.GetRequiredService<IJSRuntime>();
+        var jsRuntimeJsonSerializerOptions = jsRuntime.CloneJsonSerializerOptions();
```

Comment:
How does the `jsonOptions` parameter defined in line 34 factor into the configuration of this new JSO instance?

Reply:
That primarily gets used to deserialize event args from JavaScript events. If we reused `jsonOptions` rather than creating a new one specifically for the call to `InvokeVoidAsync()` at the end of this method, we'd have to add a `DefaultJsonTypeInfoResolver` at the end of the `TypeInfoResolverChain`. That's because developers can dynamically register their own custom `EventArgs` types that we later deserialize when that event gets emitted from JS.

However, if I do that, then startup time explodes, because [`ResolvePolymorphicConverter`](https://source.dot.net/#System.Text.Json/System/Text/Json/Serialization/JsonConverter.MetadataHandling.cs,f949d1e4ba60a9c3) ends up initializing a `JsonTypeInfo` for each of the interfaces that each of the arguments to `IJSRuntime.InvokeVoidAsync()` implement. For example, all the interfaces that `System.Int32` implement get their own `JsonTypeInfo` because we pass an `int` in the call to `InvokeVoidAsync()`.

This is probably due to the fact that `JSRuntime` serializes arguments as an `object[]`. I worked around this in an earlier prototype by instead serializing each argument individually, using `.GetType()` as the `inputType` argument in `JsonSerializer.Serialize()`. But that felt like a hack. Maybe you know of a better way to avoid this pitfall?

Reply:
I see this as another reason to deprecate the `object[]` overloads and give each argument a generic parameter.